Client/Frontend/Browser/BrowserViewController/OpenSearch/BrowserViewController+OpenSearch.swift

```diff
@@ -0,0 +1,213 @@
+
+        if searchEngineExists {
+            self.customSearchEngineButton.action = .disabled
+        } else {
+            self.customSearchEngineButton.action = .enabled
+        }
+        
+        /*
+         This is how we access hidden views in the WKContentView
+         Using the public headers we can find the keyboard accessoryView which is not usually available.
+         Specific values here are from the WKContentView headers.
+         https://github.com/JaviSoto/iOS9-Runtime-Headers/blob/master/Frameworks/WebKit.framework/WKContentView.h
+         */
+        guard let webContentView = UIView.findSubViewWithFirstResponder(webView) else {
+            /*
+             In some cases the URL bar can trigger the keyboard notification. In that case the webview isnt the first responder
+             and a search button should not be added.
+             */
+            return
+        }
+        
+        let argumentNextItem: [Any] = ["_n", "extI", "tem"]
+        let argumentView: [Any] = ["v", "ie", "w"]
+        
+        let valueKeyNextItem = argumentNextItem.compactMap { $0 as? String }.joined()
+        let valueKeyView = argumentView.compactMap { $0 as? String }.joined()
+
+        guard let input = webContentView.perform(#selector(getter: UIResponder.inputAccessoryView)),
+              let inputView = input.takeUnretainedValue() as? UIInputView,
+              let nextButton = inputView.value(forKey: valueKeyNextItem) as? UIBarButtonItem,
+              let nextButtonView = nextButton.value(forKey: valueKeyView) as? UIView else {
+            return
+        }
+        
+        inputView.addSubview(customSearchEngineButton)
+        
+        customSearchEngineButton.snp.remakeConstraints { make in
+            make.leading.equalTo(nextButtonView.snp.trailing).offset(20)
+            make.width.equalTo(inputView.snp.height)
+            make.top.equalTo(nextButtonView.snp.top)
+            make.height.equalTo(inputView.snp.height)
+        }
+    }
+
+    @objc func addCustomSearchEngineForFocusedElement() {
+        guard var reference = openSearchEngine?.reference,
+              let title = openSearchEngine?.title,
+              var url = URL(string: reference) else {
+            let alert = ThirdPartySearchAlerts.failedToAddThirdPartySearch()
+            present(alert, animated: true, completion: nil)
+            return
+        }
+                
+        guard let scheme = tabManager.selectedTab?.webView?.url?.scheme,
+              let host = tabManager.selectedTab?.webView?.url?.host else {
+            log.error("Selected Tab doesn't have URL")
+            return
+        }
+        
+        while reference.hasPrefix("/") {
+            reference.remove(at: reference.startIndex)
+        }
+        
+        let constructedReferenceURLString = "\(scheme)://\(host)/\(reference)"
```

Comment:
What do you think about limiting the schemes to `https` only?

Reply:
Fix at https://github.com/brave/brave-ios/pull/3231/commits/bfbb95a926eb3c48997c31d3bd7ab8fe2d2d3a1b